cypress/integration/plugins/security-dashboards-plugin/readonly.js

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import roleWithoutTestJson from '../../../fixtures/plugins/security-dashboards-plugin/roles/roleWithoutTest';
+import { BASE_PATH } from '../../../utils/base_constants';
+import { ADMIN_AUTH, CURRENT_TENANT } from '../../../utils/commands';
+
+const TEST_CONFIG = {
+  role: {
+    name: 'test_readonly_role',
+    json: Object.assign({}, roleWithoutTestJson, {
+      tenant_permissions: [
+        {
+          tenant_patterns: ['test_readonly_tenant'],
+          allowed_actions: ['kibana_all_read'],
+        },
+      ],
+    }),
+  },
+  tenant: {
+    name: 'test_readonly_tenant',
+    description: 'Testing read-only tenant mode',
+  },
+  user: {
+    username: 'test_readonly_user',
+    password: 'testUserPassword123',
+  },
+};
+
+if (Cypress.env('SECURITY_ENABLED')) {
+  describe('Read Only mode', () => {
+    before(() => {
+      cy.server();
+
+      cy.createTenant(TEST_CONFIG.tenant.name, {
+        description: TEST_CONFIG.tenant.description,
+      });
+
+      cy.createInternalUser(TEST_CONFIG.user.username, {
+        password: TEST_CONFIG.user.password,
+      });
+
+      cy.createRole(TEST_CONFIG.role.name, TEST_CONFIG.role.json);
+
+      cy.createRoleMapping(TEST_CONFIG.role.name, {
+        users: [TEST_CONFIG.user.username],
+      });
+    });
+
+    it('should create sample data', () => {
```

Comment:
I suppose you want to ensure the sample data is loaded for the following tests? If that's so, maybe you need to add `cy.loadSampleData('flights');` to `beforeAll`.

Reply:
@ruanyl right, I used that more for dev purposes. Fixed.